src/ast/nodes/ArrayExpression.ts

```diff
@@ -2,19 +2,24 @@ import CallOptions from '../CallOptions';
 import { ExecutionPathOptions } from '../ExecutionPathOptions';
 import {
 	arrayMembers,
+	getMemberReturnExpressionWhenCalled,
 	hasMemberEffectWhenCalled,
 	ObjectPath,
-	someMemberReturnExpressionWhenCalled
+	UNKNOWN_EXPRESSION
 } from '../values';
 import * as NodeType from './NodeType';
-import { SomeReturnExpressionCallback } from './shared/Expression';
 import { ExpressionNode, NodeBase } from './shared/Node';
 import SpreadElement from './SpreadElement';
 
 export default class ArrayExpression extends NodeBase {
 	type: NodeType.tArrayExpression;
 	elements: (ExpressionNode | SpreadElement | null)[];
 
+	getReturnExpressionWhenCalledAtPath(path: ObjectPath) {
+		if (path.length !== 1) return UNKNOWN_EXPRESSION;
```

Comment:
Sure we could. What is your intention here:
* documentation, or
* improved tree-shaking in certain situations?

I am a little hesitant to do this just in a few places as there are quite a lot of situations where we deoptimize because we would do something illegal and having it done in some places only could leave people falsely relying on this.

If you see some potential tree-shaking benefit, it might be worth doing it in all relevant places in a separate PR.